Parser.py

```diff
@@ -1,131 +0,0 @@
-            return []
-        symbol = state[symbol_index]
-        if state[symbol_index - 1] != '.' and state[len(state) - 1] is not '.':
-            return []
-        if state[symbol_index - 1] is '.':
-            state = state.replace('.', '')
-            state = state.replace(symbol, symbol + '.')
-            if state[1] is '.':
-                state = state.replace('.', '', 1)
-        return self.closure(state)
-
-    def closure(self, production):
-        state = [production]
-        point_index = production.find('.')
-        if point_index != len(production) - 1:
-            symbol = production[point_index + 1]
-            if symbol in self.grammar.get_non_terminals():
-                for i in self.grammar.get_production()[symbol]:
-                    state.append(symbol + "->." + i)
-        if len(state) == 1 and state[0][len(state[0]) - 1] is '.':
-            return [production]
-        return state
-
-    def ColCan(self):
-        s0 = self.closure(self.grammar.get_initial_state() + "'->." + self.grammar.get_initial_state())
-        self.C.append(s0)
-        while True:
-            isChanged = False
-            for state in self.C:
-                for i in state:
-                    for ind in range(len(i.split("->")[1])):
-                        index = ind + len(i.split("->")[0]) + 2
-                        if i[index] is not '.':
-                            if self.goto(i, index):
-                                if self.goto(i, index) not in self.C:
-                                    self.C.append(self.goto(i, index))
-                                    isChanged = True
-            if not isChanged:
-                self.print_C()
-                self.createTable()
-                break
-
-    def createTable(self):
-        for i in range(0,len(self.C)):
-            self.table[(i, "action")] = None
-        for state in self.C:
-            for i in state:
-                for ind in range(len(i.split("->")[1])):
-                    index = ind + len(i.split("->")[0]) + 2
-                    if i[index] is not '.':
-                        if self.goto(i, index):
-                            self.table[tuple((self.C.index(state), i[index]))] = "s" + str(
-                                self.C.index(self.goto(i, index)))
-                        else:
-                            self.table[tuple((self.C.index(state), i[index]))] = None #self.goto(i, index)
-        for i in range(0, len(self.C)):
-            resulting_states = []
-            for j in self.table.keys():
-                if j[0] == i and j[1] != "action":
-                    resulting_states.append(self.table[j])
-            for check in resulting_states:
-                if "s"+str(i) != check:
-                    self.table[(i,"action")] = "shift"
-                    break
-            if self.table[(i,"action")] != "shift":
-                 if resulting_states.count(resulting_states[0]) == len(resulting_states):
```

Comment:
Maybe clarify how "resulting states" is represented in the program and why are you counting the resulting_states[0]